app/src/main/java/com/nervousfish/nervousfish/data_objects/Profile.java

```diff
@@ -1,54 +1,79 @@
 package com.nervousfish.nervousfish.data_objects;
 
-import java.util.ArrayList;
-import java.util.Collection;
+import com.nervousfish.nervousfish.ConstantKeywords;
+
+import java.io.InvalidObjectException;
+import java.io.ObjectInputStream;
+import java.io.Serializable;
+import java.util.Arrays;
 import java.util.List;
 
 /**
- * A Profile POJO to store a name, public key and private key.
+ * A Profile POJO to store a contact representing the user and the user's keypairs.
  */
-public class Profile {
+public class Profile implements Serializable {
 
-    private final String name;
-    private final List<KeyPair> keyPairs = new ArrayList<>();
+    private static final long serialVersionUID = -4715364587956219157L;
+    private final Contact contact;
+    private final List<KeyPair> keyPairs;
 
     /**
-     * The constructor for a {@link Profile} with a single {@link KeyPair}.
+     * The constructor for the {@link Profile} class.
      *
-     * @param name the name belonging to the Profile.
-     * @param keyPair the public/private key-pair.
+     * @param contact  The contact belonging to the user.
+     * @param keyPairs the public/private key-pairs of the user.
      */
-    public Profile(final String name, final KeyPair keyPair) {
-        this.name = name;
-        this.keyPairs.add(keyPair);
+    public Profile(final Contact contact, final List<KeyPair> keyPairs) {
```

Comment:
Where should I see that? (Still don't agree though, at best you can add a method to the Profile class that transforms it into a Contact :wink:)